Iter4

F3dosik opened the pull request:
Доработанный код, который умеет принимать аргументы с использованием флагов.
Аргументы сервера:
Флаг -a=<ЗНАЧЕНИЕ> отвечает за адрес эндпоинта HTTP-сервера (по умолчанию localhost:8080).
Аргументы агента:
Флаг -a=<ЗНАЧЕНИЕ> отвечает за адрес эндпоинта HTTP-сервера (по умолчанию localhost:8080).
Флаг -r=<ЗНАЧЕНИЕ> позволяет переопределять reportInterval — частоту отправки метрик на сервер (по умолчанию 10 секунд).
Флаг -p=<ЗНАЧЕНИЕ> позволяет переопределять pollInterval — частоту опроса метрик из пакета runtime (по умолчанию 2 секунды).
При попытке передать приложению неизвестные флаги завершается с сообщением о соответствующей ошибке.
Значения интервалов времени должны задаваться в секундах.
Во всех случаях присутствуют значения по умолчанию.

F3dosik committed:
iter1: implement metric handlers

F3dosik committed:
Реализация агента для сбора и отправки метрик

- Разработан HTTP-клиент для сбора runtime-метрик
- Реализована поддержка двух типов метрик:
  - gauge (float64): 28 метрик из пакета runtime
  - counter (int64): PollCount
- Добавлены дополнительные метрики:
  - RandomValue (gauge) - случайное значение
  - PollCount (counter) - счетчик опросов
- Настроены интервалы работы:
  - pollInterval: 2s - сбор метрик
  - reportInterval: 10s - отправка на сервер
- Реализована отправка данных по HTTP POST:
  - Формат: /update/<ТИП>/<ИМЯ>/<ЗНАЧЕНИЕ>
  - Адрес: http://localhost:8080
  - Content-Type: text/plain
- Добавлена потокобезопасность через sync.RWMutex

F3dosik committed:
Использование внешних пакетов

F3dosik committed:
fix update_test.go - новая логика при использовании chi

F3dosik committed:
iter4

alekstet commented on internal/agent/sender.go:
```diff
@@ -0,0 +1,84 @@
+
+		err := s.sendMetric(metricType, metricName, metricValue)
+		if err != nil {
+			log.Printf("Ошибка отправки метрики %s: %v", metricName, err)
```
Используй уровни логгирования, а не просто Printf

alekstet commented on internal/config/agent.go:
```diff
@@ -0,0 +1,45 @@
+
+func (c *AgentConfig) Validate() error {
+	if c.Endpoint == "" {
+		return errors.New("endpoint can't be empty")
```
Не создавай каждый раз новые ошибки

alekstet commented on internal/model/metrics.go:
```diff
@@ -1,19 +1,34 @@
-	Counter = "counter"
-	Gauge   = "gauge"
+	MetricTypeCounter = "counter"
+	MetricTypeGauge   = "gauge"
```
Как вариант сделать типами

alekstet commented on internal/repository/memstorage.go:
```diff
@@ -0,0 +1,95 @@
+
+type Storage interface {
+	SetGauge(name string, value models.Gauge)
+	GetGauge(name string) (models.Gauge, error)
```
Прокидывай еще контекст

F3dosik committed:
Внесены поправки после PR:
- Оптимизация работы с ошибками
- Создание и использование нового типа MetricType для типов метрик
